src/main/drivers/video_max7456.c

```diff
@@ -0,0 +1,620 @@
+    // enable AFIO for EXTI support
+    RCC_APB2PeriphClockCmd(RCC_APB2Periph_AFIO, ENABLE);
+#endif
+
+#ifdef STM32F303xC
+    /* Enable SYSCFG clock otherwise the EXTI irq handlers are not called */
+    RCC_APB2PeriphClockCmd(RCC_APB2Periph_SYSCFG, ENABLE);
+#endif
+
+#ifdef STM32F10X
+    gpioExtiLineConfig(extiConfig->exti_port_source, extiConfig->exti_pin_source);
+#endif
+
+#ifdef STM32F303xC
+    gpioExtiLineConfig(extiConfig->exti_port_source, extiConfig->exti_pin_source);
+#endif
+
+    registerExtiCallbackHandler(extiConfig->exti_irqn, handlerFn);
+
+    EXTI_ClearITPendingBit(extiConfig->exti_line);
+
+    EXTI_InitTypeDef EXTIInit;
+    EXTIInit.EXTI_Line = extiConfig->exti_line;
+    EXTIInit.EXTI_Mode = EXTI_Mode_Interrupt;
+    EXTIInit.EXTI_Trigger = trigger;
+    EXTIInit.EXTI_LineCmd = ENABLE;
+    EXTI_Init(&EXTIInit);
+}
+
+void max7456_extiConfigure(
+    const extiConfig_t *losExtiConfig,
+    const extiConfig_t *vSyncExtiConfig,
+    const extiConfig_t *hSyncExtiConfig
+)
+{
+    max7456LOSExtiConfig = losExtiConfig;
+    max7456VSYNCExtiConfig = vSyncExtiConfig;
+    max7456HSYNCExtiConfig = hSyncExtiConfig;
+
+    max7456_extiInit(max7456LOSExtiConfig, LOS_EXTI_Handler, EXTI_Trigger_Rising_Falling);
+    max7456_extiInit(max7456VSYNCExtiConfig, VSYNC_EXTI_Handler, EXTI_Trigger_Falling);
+    max7456_extiInit(max7456HSYNCExtiConfig, HSYNC_EXTI_Handler, EXTI_Trigger_Falling);
+}
+
+textScreen_t *max7456_getTextScreen(void)
+{
+    return &max7456Screen;
+}
+
+static void max7456_write(uint8_t address, uint8_t data)
+{
+    ENABLE_MAX7456;
+
+    spiTransferByte(MAX7456_SPI_INSTANCE, address);
+    spiTransferByte(MAX7456_SPI_INSTANCE, data);
+
+    DISABLE_MAX7456;
+}
+
+static uint8_t max7456_read(uint8_t address)
+{
+    uint8_t result;
+
+    ENABLE_MAX7456;
+
+    spiTransferByte(MAX7456_SPI_INSTANCE, address);
+    result = spiTransferByte(MAX7456_SPI_INSTANCE, 0xff);
+
+    DISABLE_MAX7456;
+
+    return result;
+}
+
+static void max7456_setVideoMode(videoMode_e videoMode)
+{
+    switch(videoMode)
+    {
+        case VIDEO_NTSC:
+            max7456_videoModeMask = MAX7456_MODE_MASK_NTSC;
+            max7456Screen.height = MAX7456_NTSC_ROW_COUNT;
+            break;
+         case VIDEO_PAL:
+            max7456_videoModeMask = MAX7456_MODE_MASK_PAL;
+            max7456Screen.height = MAX7456_PAL_ROW_COUNT;
+            break;
+    }
+
+    max7456Screen.width = MAX7456_COLUMN_COUNT;
+}
+
+bool max7456_isOSDEnabled(void)
+{
+    uint8_t result = max7456_read(MAX7456_REG_VM0_READ);
+    //debug[3] = result;
+
+    bool osdEnabled = (result & MAX7456_VM0_BIT_OSD_ENABLE) != 0;
+
+    return osdEnabled;
+}
+
+static bool max7456_isResetComplete(void)
+{
+    uint8_t result = max7456_read(MAX7456_REG_STAT_READ);
+    //debug[3] = result;
+
+    bool resetComplete = (result & MAX7456_STAT_BIT_RESET_MODE) == 0;
+
+    return resetComplete;
+}
+
+static void max7456_softReset()
+{
+    static uint32_t resetWait = 0;
```

Comment:
well yes/no.  I would like some way of detecting max7456 communication failures and to initiate a soft reset, followed by a hard reset if still no communication.